logstreams/src/main/java/io/zeebe/logstreams/processor/StreamProcessorController.java

```diff
@@ -145,9 +116,9 @@ protected void onActorStarting() {
 
     logStreamReader.wrap(logStream);
     logStreamWriter.wrap(logStream);
+    logStreamWriter.producerId(streamProcessorContext.getId());
```

Comment:
The producer id needs to be set every time because it is reset. If the producer id is not set then we may have problems on reprocessing.